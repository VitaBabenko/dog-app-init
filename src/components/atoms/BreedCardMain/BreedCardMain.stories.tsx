import type { Meta, StoryObj } from '@storybook/react';
import { BreedCardMain } from '.';

const meta = {
  title: 'Atoms/BreedCardMain',
  component: BreedCardMain
} satisfies Meta<typeof BreedCardMain>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    id: '',
    url: '',
    breeds: [
      {
        id: 1,
        name: ''
      }
    ],
    favouriteId: ''
  }
};
