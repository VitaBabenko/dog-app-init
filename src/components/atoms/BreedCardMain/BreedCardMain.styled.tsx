import { Button, styled } from '@mui/material';

export const StyledButton = styled(Button)({
  display: 'flex',
  gap: '10px',
  position: 'absolute',
  bottom: '10px',
  left: '50%',
  transform: 'translate(-50%)',
  width: '245px',
  padding: '10px 20px'
});
