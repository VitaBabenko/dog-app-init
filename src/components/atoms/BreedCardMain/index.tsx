import { IconButton } from '@mui/material';
import { FC, useState } from 'react';
import { Link } from 'react-router-dom';
import { CardWrapper } from '../CardWrapper';
import { IconHeart } from '../IconHeart';
import {
  useAddFavouritesMutation,
  useDeleteFavouritesMutation
} from '../../../services/favourites';
import { IconDogFoot } from '../IconDogFoot';
import { ArrowRight } from '../ArrowRight';
import { StyledButton } from './BreedCardMain.styled';

type BreedCardMainProps = {
  id: string;
  url: string;
  breeds: [
    {
      id: number;
      name: string;
    }
  ];
  favouriteId: string | null;
};

export const BreedCardMain: FC<BreedCardMainProps> = ({
  id,
  url,
  breeds,
  favouriteId
}) => {
  const [addFavourites] = useAddFavouritesMutation();
  const [deleteFavourites] = useDeleteFavouritesMutation();
  const [isHoveredCard, setIsHoveredCard] = useState<boolean>(false);

  const handleMouseOver = () => setIsHoveredCard(true);
  const handleMouseOut = () => setIsHoveredCard(false);

  const handleBtnAdd = (breedId: string) => () => {
    console.log('add favourite');
    addFavourites({ image_id: breedId });
  };

  const handleBtnDelete = (breedId: string) => () => {
    console.log('delete');
    deleteFavourites({ id: breedId });
  };

  return (
    <CardWrapper
      key={id}
      variant="primary"
      borderRadius={0}
      onMouseOver={handleMouseOver}
      onMouseOut={handleMouseOut}
    >
      <img src={url} alt={id} height="100%" />
      {favouriteId !== null ? (
        <IconButton
          sx={{
            position: 'absolute',
            top: '10px',
            right: '10px'
          }}
          onClick={handleBtnDelete(favouriteId)}
        >
          <IconHeart state="active" />
        </IconButton>
      ) : (
        <IconButton
          sx={{
            position: 'absolute',
            top: '10px',
            right: '10px'
          }}
          onClick={handleBtnAdd(id)}
        >
          {isHoveredCard ? (
            <IconHeart state="hover" />
          ) : (
            <IconHeart state="default" />
          )}
        </IconButton>
      )}
      {isHoveredCard && breeds.length === 1 && (
        <Link to={`breeds/${breeds[0].id}`}>
          <StyledButton variant="contained" color="secondary">
            <IconDogFoot state="secondary" />
            {breeds[0].name}
            <ArrowRight />
          </StyledButton>
        </Link>
      )}
    </CardWrapper>
  );
};
