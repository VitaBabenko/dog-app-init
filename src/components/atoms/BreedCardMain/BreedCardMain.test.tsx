import { composeStories } from '@storybook/react';
import * as stories from './BreedCardMain.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('BreedCardMain', () => {
  const { Primary } = composeStories(stories);
  testSnapshot(<Primary />);
});
