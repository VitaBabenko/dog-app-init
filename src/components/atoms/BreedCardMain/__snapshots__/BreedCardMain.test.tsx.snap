// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreedCardMain should match snapshot Component 1`] = `
.emotion-0 {
  border-radius: 0;
  position: relative;
  cursor: pointer;
  -webkit-background-position: center;
  background-position: center;
  -webkit-background-size: cover;
  background-size: cover;
  background-repeat: no-repeat;
  overflow: hidden;
  max-width: 590px;
  min-height: 90px;
  background-color: #FFF;
  box-shadow: 6px 6px 0px 0px #000000;
  border: 1px solid #090808;
  border-radius: 20px;
}

.emotion-0 img {
  display: block;
  width: 100%;
  object-fit: cover;
}

.emotion-0:hover,
.emotion-0:active {
  box-shadow: 6px 6px 8px 0px #921FED;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 2.142857142857143rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: none;
  box-shadow: none;
  position: absolute;
  top: 10px;
  right: 10px;
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-1:hover {
  background-color: none;
  border: none;
  box-shadow: none;
}

.emotion-1:active,
.emotion-1:focus {
  background-color: none;
  border: none;
  box-shadow: none;
}

<div>
  <div
    class="MuiBox-root emotion-0"
    variant="primary"
  >
    <img
      alt=""
      height="100%"
      src=""
    />
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-1"
      tabindex="0"
      type="button"
    >
      <test-file-stub
        aria-hidden="true"
        classname="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1v8usww-MuiSvgIcon-root"
        focusable="false"
      />
    </button>
  </div>
</div>
`;
